WordPressVIPMinimum/Sniffs/Performance/LowExpiryCacheTimeSniff.php

```diff
@@ -69,21 +81,34 @@ public function process_parameters( $stackPtr, $group_name, $matched_content, $p
 			return;
 		}
 
-		$time = $parameters[4]['raw'];
-
-		if ( false === is_numeric( $time ) ) {
-			// If using time constants, we need to convert to a number.
-			$time = str_replace( array_keys( $this->wp_time_constants ), $this->wp_time_constants, $time );
+		// If using time constants, we need to convert to a number.
+		$time = str_replace( array_keys( $this->wp_time_constants ), $this->wp_time_constants, $parameters[4]['raw'] );
 
-			if ( preg_match( '#^[\s\d+*/-]+$#', $time ) > 0 ) {
-				$time = eval( "return $time;" ); // phpcs:ignore Squiz.PHP.Eval -- No harm here.
+		$rand_function = false;
+		foreach ( $this->rand_functions as $fn ) {
+			if ( false !== strpos( $time, $fn ) ) {
+				$rand_function = $fn;
+				break;
 			}
 		}
 
-		if ( $time < 300 ) {
-			$message = 'Low cache expiry time of "%s", it is recommended to have 300 seconds or more.';
-			$data    = [ $parameters[4]['raw'] ];
-			$this->phpcsFile->addWarning( $message, $stackPtr, 'LowCacheTime', $data );
+		$times = [];
+		if ( false !== $rand_function ) {
+			$times = explode( ',', preg_replace( '/[( )|\(|\)|(' . $rand_function . ')]/', '', $time ) );
```

Comment:
I think this regex is going to miss cases where there are unusual code styling - line breaks before, between, or after string arguments like `20 * HOUR_IN_SECONDS` - I'd at least like to see more unit tests, and I think when we switch to be able to use https://phpcsutils.com/phpdoc/classes/PHPCSUtils-Utils-PassedParameters.html#method_getParameters then this will be much easier to be confident with.